rustls/src/server/tls12.rs

```diff
@@ -273,14 +273,35 @@ mod client_hello {
 
         fn pick_kx_group(
             &self,
-            groups_ext: &[NamedGroup],
+            groups_ext: Option<&[NamedGroup]>,
         ) -> Option<&'static dyn SupportedKxGroup> {
-            self.config
-                .provider
-                .kx_groups
-                .iter()
-                .find(|skxg| groups_ext.contains(&skxg.name()))
-                .cloned()
+            let groups_ext_or_default = groups_ext.unwrap_or(&[]);
```

Comment:
Style suggestions:

* Rename `groups_ext_or_default` to `peer_kx_groups` -- as I understand it this represents the counterpart to `our_kx_groups` (and the `or_default` suffix doesn't really make sense in terms of a value IMO)
* IMO multi-line combinator expressions in `if` conditions make the code hard to follow -- probably better to name those values which should also make it clearer what's going on
* Would be good to invert the last branch so we can yield `None` early and reduce rightward drift in picking a group from `our_kx_groups`

Reply:
Suggestions applied!